antarest/study/business/bindingconst_management.py

```diff
@@ -0,0 +1,217 @@
+        binding_constraint = []
+        for config_value in config.values():
+            new_config: BindingConstDTO = BindingConstDTO(
+                id=config_value["id"],
+                name=config_value["name"],
+                enabled=config_value["enabled"],
+                time_step=TimeStep.from_str(config_value["type"]),
+                operator=config_value["operator"],
+                comments=config_value["comments"] if "comments" in config_value else None,
+                constraints=None
+            )
+
+            for key, value in config_value.items():
+                split = key.split("%")
+                if len(split) == 2:
+                    area1 = split[0]
+                    area2 = split[1]
+                    weight = 0.0
+                    offset = None
+                    try:
+                        weight = float(value)
+                    except ValueError:
+                        value_split = value.split("%")
+                        if len(value_split) == 2:
+                            weight = value_split[0]
+                            offset = value_split[1]
+                    if new_config.constraints is None:
+                        new_config.constraints = {}
+                    new_config.constraints[key] = ConstraintDTO(
+                        id=key,
+                        weight=weight,
+                        offset=offset if offset is not None else None,
+                        data=LinkInfoDTO(
+                            area1=area1,
+                            area2=area2,
+                        ))
+                    continue
+
+                split = key.split(".")
+                if len(split) == 2:
+                    area = split[0]
+                    cluster = split[1]
+                    weight = 0
+                    offset = None
+                    try:
+                        weight = float(value)
+                    except ValueError:
+                        value_split = value.split("%")
+                        if len(value_split) == 2:
+                            weight = value_split[0]
+                            offset = value_split[1]
+                    if new_config.constraints is None:
+                        new_config.constraints = {}
+                    new_config.constraints[key] = ConstraintDTO(
+                        id=key,
+                        weight=weight,
+                        offset=offset if offset is not None else None,
+                        data=ClusterInfoDTO(
+                            area=area,
+                            cluster=cluster,
+                        ))
+                    continue
+
+            binding_constraint.append(new_config)
+        return binding_constraint
+
+    def get_binding_constraint(self, study: Study, binding_constraint_id: str) -> Optional[BindingConstDTO]:
+        bc = self.get_binding_constraint_list(study)
```

Comment:
parsing all constraints to get only one is not great.
maybe add an additional param "constraint_id" to get_binding_constraint_list so that it will return only the asked binding_constraint instead of parsing them all